hibernate-core/src/test/java/org/hibernate/internal/util/ReflectHelperTest.java

```diff
@@ -230,4 +230,15 @@ public void test_getMethod_nestedInterfaces_on_superclasses()
 	public void test_setMethod_nestedInterfaces_on_superclasses() {
 		assertNotNull( ReflectHelper.findSetterMethod( E.class, "id", String.class ) );
 	}
+
+	@TestForIssue(jiraKey = "HHH-14059")
+	@Test
+	public void test_getConstantValue_UpperCaseEnum() {
+		when( sessionFactoryOptionsMock.isConventionalJavaConstants() ).thenReturn( true );
+
+		when( classLoaderServiceMock.classForName( "com.example.UStatus" ) ).thenReturn( (Class) Status.class );
+		Object value = ReflectHelper.getConstantValue( "com.example.UStatus.OFF", sessionFactoryImplementorMock);
+		assertEquals( OFF, value );
+		verify(classLoaderServiceMock, times(1)).classForName( eq("com.example.UStatus") );
+	}
```

Comment:
In Hibernate testing, we seldom rely on mocking but go about testing through real integration test.

Reply:
Hi @NathanQingyangXu ,  thank you for your comment. I found I can extend `BaseEntityManagerFunctionalTestCase` to get an `entityManagerFactory` so that there is no need for mocking. However, the existing tests of this `ReflectHelper` use mocking. I want to be consistent with the [existing](https://github.com/hibernate/hibernate-orm/blob/1fcfbaa1c070846ba075ba3e40cbeceba7bf8047/hibernate-core/src/test/java/org/hibernate/internal/util/ReflectHelperTest.java).